hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/Util.java

```diff
@@ -246,6 +257,89 @@ public static void checkSerializable(Object object, String objectName) {
         return res;
     }
 
+    public static void copyStream(InputStream in, OutputStream out) throws IOException {
+        byte[] buf = new byte[BUFFER_SIZE];
+        for (int readCount; (readCount = in.read(buf)) > 0; ) {
+            out.write(buf, 0, readCount);
+        }
+    }
+
+    public static void unzip(InputStream is, Path targetPath) throws IOException {
+        try (ZipInputStream zipIn = new ZipInputStream(is)) {
+            for (ZipEntry ze; (ze = zipIn.getNextEntry()) != null; ) {
+                String entryName = ze.getName();
+                Path destPath = targetPath.resolve(entryName);
+                if (ze.isDirectory()) {
+                    Files.createDirectory(destPath);
+                } else {
+                    Path parent = destPath.getParent();
+                    if (parent != null) {
+                        Files.createDirectories(parent);
+                        try (OutputStream fileOut = Files.newOutputStream(destPath)) {
+                            copyStream(zipIn, fileOut);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates a ZIP-file stream from the directory tree rooted at the supplied
+     * {@code baseDir}. Pipes the stream into the provided output stream, closing
+     * it when done.
+     * <p>
+     * <strong>Note:</strong> files and directories starting with either {@code _}
+     * or {@code .} are ignored.
```

Comment:
- This should be documented at `JobConfig.attachDirectory`, not here. 
- It is a non-portable way to detect hidden files, it's valid on macOS only i guess (never heard about the `_` prefix). You should use `Files.isHidden(path)`.
- You should also ignore hidden files, not just hidden directories.
- It might be easier to use `Files.find`

Reply:
I've partially fixed this. Javadoc still needs updating.